homeassistant/components/radiotherm/climate.py

```diff
@@ -98,251 +114,187 @@ def round_temp(temperature):
 
-    async def async_added_to_hass(self):
+    async def async_added_to_hass(self) -> None:
         """Register callbacks."""
         # Set the time on the device.  This shouldn't be in the
         # constructor because it's a network call.  We can't put it in
         # update() because calling it will clear any temporary mode or
         # temperature in the thermostat.  So add it as a future job
         # for the event loop to run.
         self.hass.async_add_job(self.set_time)
+        await super().async_added_to_hass()
 
-    @property
-    def name(self):
-        """Return the name of the Radio Thermostat."""
-        return self._name
-
-    @property
-    def temperature_unit(self):
-        """Return the unit of measurement."""
-        return TEMP_FAHRENHEIT
-
-    @property
-    def precision(self):
-        """Return the precision of the system."""
-        return PRECISION_HALVES
-
-    @property
-    def extra_state_attributes(self):
-        """Return the device specific state attributes."""
-        return {ATTR_FAN_ACTION: self._fstate}
-
-    @property
-    def fan_modes(self):
-        """List of available fan modes."""
-        if self._is_model_ct80:
-            return CT80_FAN_OPERATION_LIST
-        return CT30_FAN_OPERATION_LIST
-
-    @property
-    def fan_mode(self):
-        """Return whether the fan is on."""
-        return self._fmode
-
-    def set_fan_mode(self, fan_mode):
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Turn fan on/off."""
-        if (code := FAN_MODE_TO_CODE.get(fan_mode)) is not None:
-            self.device.fmode = code
-
-    @property
-    def current_temperature(self):
-        """Return the current temperature."""
-        return self._current_temperature
-
-    @property
-    def current_humidity(self):
-        """Return the current temperature."""
-        return self._current_humidity
-
-    @property
-    def hvac_mode(self) -> HVACMode:
-        """Return the current operation. head, cool idle."""
-        return self._current_operation
-
-    @property
-    def hvac_action(self) -> HVACAction | None:
-        """Return the current running hvac operation if supported."""
-        if self.hvac_mode == HVACMode.OFF:
-            return None
-        return self._tstate
-
-    @property
-    def target_temperature(self):
-        """Return the temperature we try to reach."""
-        return self._target_temperature
-
-    @property
-    def preset_mode(self):
-        """Return the current preset mode, e.g., home, away, temp."""
-        if self._program_mode == 0:
-            return PRESET_HOME
-        if self._program_mode == 1:
-            return PRESET_ALTERNATE
-        if self._program_mode == 2:
-            return PRESET_AWAY
-        if self._program_mode == 3:
-            return PRESET_HOLIDAY
+        if (code := FAN_MODE_TO_CODE.get(fan_mode)) is None:
+            raise HomeAssistantError(f"{fan_mode} is not a valid fan mode")
```

Comment:
We normally raise ValueError on invalid user input and HomeAssistantError on all other errors in service handlers.

Reply:
Will adjust tomorrow after sleep

Reply:
Fixed in https://github.com/home-assistant/core/pull/73104